### DISCUSSION TITLE
How to get phone number on first incoming message?

### DISCUSSION THREAD

**Discussion by rez0n** [AsamK/signal-cli#1414] [Q&A] [answered]
Hi All,

When signal-cli receive first message from new contact the phone number of the sender is not available, only uuid.
I need to have phone number on first incoming message.
As I read in different sources it is built-in Signal privacy feature and it seems impossible to overcome, but just in case maybe signal-cli have option to deal with this?
I found `--trust-new-identities=always` option, but it seems could not help in this case.
I can't test it much because it reproduces only once per new phone numbers combination.

Thanks in advance!

**Comment by AsamK** [comment, marked as answer]
The Signal-Server used to include the phonenumber in the first message, but now it doesn't and there's no way to overcome this.
The phonenumber is only included in unidentified sender messages, which by default require the profile key and thus only work after the first message exchange.

There is however an option to allow unidentified sender messages from everyone (see [Signal blog post](https://signal.org/blog/sealed-sender/#special-delivery)), that would allow the first message to you to be an unidentified sender message. Then you should get the number with the first message already. Only exception would be users that have enabled phone number privacy and don't send their phone number at all.
signal-cli doesn't provide an option for this yet, but it shouldn't be hard to add that.

**Reply by AsamK** [reply]
I've added a new parameter for this.
With the [latest development build](https://github.com/AsamK/signal-cli/actions/runs/7513169103) you can try to run this command on the account where you want to receive messages:
`signal-cli -a +XXX updateAccount --unrestricted-unidentified-sender=true`

Then try sending to that account from a new phone number.

**Reply by rez0n** [reply]
Hi @AsamK 
Thank you, I tested and it works perfect! Much appreciated.

Is there any release schedule? (I'm asking just for information to "calculate" rough timing of build of the standalone apt package generated by [packaging.gitlab.io](https://packaging.gitlab.io/signal-cli/) project because I need ARM64 package for "production" environment)

**Reply by AsamK** [reply]
That's great, thanks for testing!

There's no release schedule, but there should be a new release in the next few weeks.

**Reply by mattwr18** [reply]
I saw this was released in v0.13.0. I ran into the same issue as the OP and maybe I'm missing something, but this doesn't seem to work for me. I'm on the latest version v0.13.4 and I've run the command :point_up: and I still only receive the uuid if I have not sent a message to the phone number.

Here is the verbose output of the command. I don't see any failures. Although it doesn't return anything that lets me know it is successful and I don't see a way to query for an accounts settings. I can open an issue, or maybe I'm missing something obvious? 

```
signal-cli --config /home/.local/share/signal-cli -v -a ACCOUNT updateAccount --unrestricted-unidentified-sender=true 
2024-06-14T07:43:58.519Z [main] DEBUG org.asamk.signal.util.IOUtils - XDG_RUNTIME_DIR not set, falling back to temp dir
2024-06-14T07:43:58.552Z [main] DEBUG org.asamk.signal.App - Starting signal-cli 0.13.4
2024-06-14T07:43:58.754Z [main] INFO  LibSignal - [libsignal]: rust/bridge/jni/src/logging.rs:161: Initializing libsignal version:0.47.0
2024-06-14T07:43:59.252Z [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
2024-06-14T07:43:59.426Z [main] INFO  com.zaxxer.hikari.pool.HikariPool - HikariPool-1 - Added connection org.sqlite.jdbc4.JDBC4Connection@6bea52d4
2024-06-14T07:43:59.429Z [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
2024-06-14T07:44:02.058Z [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown initiated...
2024-06-14T07:44:02.064Z [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Shutdown completed.
```
